Kernel/Locking/Mutex.cpp

```diff
@@ -323,18 +323,15 @@ void Mutex::restore_lock(Mode mode, u32 lock_count, [[maybe_unused]] LockLocatio
     VERIFY(mode != Mode::Unlocked);
     VERIFY(lock_count > 0);
     VERIFY(!Processor::current_in_irq());
-    auto current_thread = Thread::current();
+    auto* current_thread = Thread::current();
     bool did_block = false;
     SpinlockLocker lock(m_lock);
     switch (mode) {
     case Mode::Exclusive: {
         auto previous_mode = m_mode;
-        bool need_to_block = false;
-        if (m_mode == Mode::Exclusive && m_holder != current_thread)
-            need_to_block = true;
-        else if (m_mode == Mode::Shared && (m_shared_holders.size() != 1 || !m_shared_holders.contains(current_thread)))
-            need_to_block = true;
-        if (need_to_block) {
+        if ((m_mode == Mode::Exclusive && m_holder != current_thread)
+            || (m_mode == Mode::Shared && m_shared_holders.size() != 1)
+            || !m_shared_holders.contains(current_thread)) {
```

Comment:
This condition might be slightly different.
Before it is `m_mode == Mode::Shared && (m_shared_holders.size() != 1 || !m_shared_holders.contains(current_thread))`
So `m_mode == Mode::Shared` must be true and at least one of `m_shared_holders.size() != 1 || !m_shared_holders.contains(current_thread)` must be true.
But now if
`!m_shared_holders.contains(current_thread)` is true without `m_mode == Mode::Shared` being true it still takes the branch.

Reply:
Yeah that was the thing that borked CI
It made locking not working and blocked...